src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts

```diff
@@ -298,7 +298,28 @@ export class LearnPageComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   public viewAll(event) {
-    const searchQuery = JSON.parse(event.searchQuery);
+    const userData = this.userService.userProfile;
+    const requestSearchQuery = JSON.stringify({
```

Comment:
Change the label requestSearchQuery -> myCoursesSearchQuery

Reply:
Define as private level const. Easy to change & find

Reply:
Is Imperssion event is generating for view-All button click?
Section details has to go init

Reply:
interact after clicking the view-all button and impression after landing in view-all page, both are generated.

Reply:
done